server/channels/app/post.go

```diff
@@ -1338,6 +1350,10 @@ func (a *App) DeletePost(c request.CTX, postID, deleteByID string) (*model.Post,
 		a.deleteFlaggedPosts(post.Id)
 	})
 
+	if err = a.RemoveNotifications(c, post, channel, team); err != nil {
```

Comment:
I guess this is handled in a different way, but just as sanity check...

If the root post is removed, none of this is needed? I guess the root post getting deleted, sets the thread as deleted and doesn't sum for mentions count, but just confirming.